webpack/hashed-folder-copy-plugin/src/HashedFolderCopyPlugin.ts

```diff
@@ -267,21 +267,19 @@ export class HashedFolderCopyPlugin implements webpack.Plugin {
         resolvedGlobsBase = path.join(packagePath, pathInsidePackage);
       }
 
-      for (const globPattern of globPatterns) {
-        const globResults: string[] = glob.sync(globPattern, { cwd: resolvedGlobsBase, nodir: true });
-        for (const globResult of globResults) {
-          const globResultFullPath: string = path.resolve(resolvedGlobsBase, globResult);
+      const globResults: string[] = glob.sync(globPatterns, { cwd: resolvedGlobsBase, onlyFiles: true });
+      for (const globResult of globResults) {
+        if (assetsToAdd.has(globResult)) {
+          const errorMessage: string = `Two files resolve to the same output path "${globResult}"`;
+          compilation.errors.push(new Error(errorMessage));
+          return this._renderError(errorMessage);
+        }
 
-          if (assetsToAdd.has(globResult)) {
-            const errorMessage: string = `Two files resolve to the same output path "${globResult}"`;
-            compilation.errors.push(new Error(errorMessage));
-            return this._renderError(errorMessage);
-          }
+        const globResultFullPath: string = path.resolve(resolvedGlobsBase, globResult);
```

Comment:
Should we have the globber return absolute paths so we don't have to do this all the time?

Reply:
The relative path is used separately in this function, so in this scenario it is best to defer the resolving.